src/clients/BaseAbstractClient.ts

```diff
@@ -37,6 +51,34 @@ export abstract class BaseAbstractClient {
     this._isUpdated = value;
   }
 
+  /**
+   * Validates and updates the stored EventSearchConfig in advance of an update() call.
+   * Use isEventSearchConfig() to discriminate the result.
+   * @provider Ethers RPC provider instance.
+   * @returns An EventSearchConfig instance if valid, otherwise an UpdateFailureReason.
+   */
+  public async updateSearchConfig(provider: providers.Provider): Promise<EventSearchConfig | UpdateFailureReason> {
+    const fromBlock = this.firstBlockToSearch;
+    let { toBlock } = this.eventSearchConfig;
+    if (isDefined(toBlock)) {
+      if (fromBlock > toBlock) {
+        return UpdateFailureReason.BadRequest;
+      }
+    } else {
+      try {
+        toBlock = await provider.getBlockNumber();
```

Comment:
Would this result in failure cases where we send 2x the # of rpc requests per failure? in other words, do you see this  extra rpc request making failures even noisier?

Reply:
I don't think it sends any more requests than the current implementation, but I might be missing a scenario. At present the various clients typically issue the `getBlockNumber()` call on each update because in most cases `eventSearchConfig.toBlock` is not set.